### PR TITLE
Fix RequestInit clone body preservation

### LINKED ISSUE #4686: Bug: Request body is dropped when cloning with options
### Summary
When creating a new `Request` from an existing `Request` with `RequestInit` options (e.g., adding headers), the original body is lost if `options.body` is not provided. This causes POST/PUT payloads to be silently dropped.

### To Reproduce
```javascript
const original = new Request("http://unit.test", {
  method: "POST",
  body: "payload",
});

const withHeaders = new Request(original, {
  headers: { "x-test": "1" },
});

fetch(withHeaders);
```

### Expected behavior
The body `"payload"` is preserved when cloning a `Request` and only modifying headers/methods.

### Actual behavior
The request body becomes empty when any `RequestInit` is provided without an explicit `body`.

### Root cause (code pointers)
The body is dropped in `RequestInit::into_request_builder` when reconstructing the request from parts. The existing body is intentionally discarded:
- `core/runtime/src/fetch/request.rs` (RequestInit::into_request_builder)
- The clone path uses this method via `JsRequest::create_from_js`.

### Why this is critical
Any user who needs to add headers (auth, content-type, tracing) to an existing `Request` with a body will unintentionally send empty payloads, breaking common Fetch workflows.

### Environment
- Boa version: main branch (latest)
- OS: macOS (also likely cross-platform)

### PR DESCRIPTION
**Pull request by Nakshatra480**
Closes #4686 

## Summary
- Preserve the original request body when cloning a Request with RequestInit options unless `body` is explicitly provided.
- Add regression tests for Request cloning edge cases (empty body, override body, no body).

## Changes
- Keep the existing body from `HttpRequest::into_parts()` and only overwrite it when `RequestInit.body` is set.
- Add unit tests covering cloning behavior with empty/overridden/no body.

## Testing
- `cargo test -p boa_runtime request_clone_`
- `cargo fmt -- --check`
- `cargo clippy --all-features --all-targets`

### CODE REVIEW

**Comment by Nakshatra480** [comment]
Hi @jedel1043, i have solved the issue. Can you pls review this PR.
Thanks for your time😊

**Comment by Nakshatra480** [comment]
Hey @jedel1043  just wanted to bump this PR.
It fixes an issue where cloning a Request with RequestInit could accidentally drop the original body. Now the body is preserved unless RequestInit.body is explicitly provided.

The implementation keeps the body from into_parts() and only overrides it when a new one is passed in. I also added regression tests covering the edge cases around cloning and body overrides.

Would really appreciate a review whenever someone from the runtime side has time. Happy to make any adjustments needed 🙂

File: core/runtime/src/fetch/request.rs [hunk 1]
```diff
@@ -66,8 +66,9 @@ impl RequestInit {
         request: Option<HttpRequest<Vec<u8>>>,
     ) -> JsResult<HttpRequest<Vec<u8>>> {
         let mut builder = HttpRequest::builder();
+        let mut request_body = None;
```
Comment (jedel1043) [inline core/runtime/src/fetch/request.rs]: This kinda carried over from the original implementation, but `Vec::new` can already act as our default value. This way we can skip all the `Some` wrapping and the `unwrap_or_default` call at the end.

Comment (Nakshatra480) [reply core/runtime/src/fetch/request.rs]: Thanks for the review! I've addressed this by using Vec::new() directly as the default value for request_body. This successfully removed the need for Option/Some wrapping and the unwrap_or_default() call at the end, making it much cleaner.

**Comment by Nakshatra480** [comment]
> Nice catch! I have a small nitpick but everything else looks good!

@jedel1043 i have done the suggested changes, can you review this PR now.
Thanks for your feedback
